lib/expressView.js

```diff
@@ -20,7 +20,30 @@ var format = require('util').format;
 var inherit = require('util').inherits;
 var debug = require('debug')(require('../package').name);
 
-var View = util.safeRequire('express/lib/view');
+var ExpressView = util.safeRequire('express/lib/view');
+
+function View(name, options) {
+  options = options || {};
+  this.name = name;
+  this.root = options.root;
+  var engines = options.engines;
+  this.defaultEngine = options.defaultEngine;
+  console.log(this.defaultEngine);
+
+  //var ext = this.ext = extname(name);
```

Comment:
The stock `View` from express takes the `name` and checks it for a file extension. made for getting the `.ejs` part of `index.ejs` (uses [`path.extname`](https://nodejs.org/api/path.html#path_path_extname_p)). We want to get rid of this original functionality because it means that when have a `name` (aka url) that has a period the `View` will try to get the extname for that variable. So `hello?foo=example.com` will think the `ext` is `.com`. This is bad because the `ext` is tied to the rendering engine. So the normal `View` will think that the rendering engine is `.com` instead of `.js` or `.jsx`, thus the error we get when we have periods in the name (url).